src/dqe.erl

```diff
@@ -142,407 +154,100 @@ run(Query, Timeout) ->
-    lists:flatten(Parts1).
-
-expand_part({named, N, P}, Buckets) ->
-    %% TODO adjust the name here!
-    [{named, update_name(N, M, G), P1}
-     || {M, G, P1} <- expand_part(P, Buckets)];
-expand_part({combine, Fun, Parts}, Buckets) ->
-    Parts1 = [P || {_, _, P} <- expand_parts(Parts, Buckets)],
-    [{keep, keep, {combine, Fun, Parts1}}];
-expand_part({calc, C, {get, {B, M}}}, _Buckets) ->
-    %% We convert the metric from a list to a propper metric here
-    [{keep, keep, {calc, C, {get, {B, dproto:metric_from_list(M)}}}}];
-expand_part({calc, C, {combine, _, _}= Comb} , Buckets) ->
-    %% We convert the metric from a list to a propper metric here
-    [{keep, keep, Comb1}] = expand_part(Comb, Buckets),
-    [{keep, keep, {calc, C, Comb1}}];
-
-expand_part({calc, C, {sget, {B, G}}}, Buckets) ->
-    Ms = orddict:fetch(B, Buckets),
-    {ok, Selected} = dqe:glob_match(G, Ms),
-    [{{metric, M}, {glob, G}, {calc, C, {get, {B, M}}}} || M <- Selected];
-
-expand_part({calc, C, {lookup, Query}}, _Buckets) ->
-    {ok, Selected} = dqe_idx:lookup(Query),
-    pdebug('prepare', "Looked up ~p metrics for ~p.",
-           [length(Selected), Query]),
-    [{{get, {B, M}}, {lookup, Query}, {calc, C, {get, {B, M}}}}
-     || {B, M} <- Selected].
-
-update_name(Name, keep, keep) ->
-    Name;
-update_name(Name, {metric, Metric}, {glob, Glob}) ->
-    MList = dproto:metric_to_list(Metric),
-    GStr = dql:unparse_metric(Glob),
-    MStr = dql:unparse_metric(MList),
-    binary:replace(Name, GStr, MStr);
-
-update_name(Name, {get, {Bucket, Metric}}, Lookup = {lookup, _}) ->
-    MList = dproto:metric_to_list(Metric),
-    LStr = dql:unparse(Lookup),
-    GStr = dql:unparse({get, {Bucket, MList}}),
-    binary:replace(Name, LStr, GStr).
-
-name_parts([Q | R], Acc, Aliases, Buckets) ->
-    case name(Q, Aliases, Buckets) of
-        {ok, {Name, Translated}} ->
-            Q1 = {dqe_name, [Name, Translated]},
-            name_parts(R, [Q1 | Acc], Aliases, Buckets);
-        E ->
-            E
-    end;
-
-name_parts([], Acc, _Aliases, _Buckets) ->
-    {ok, lists:reverse(Acc)}.
 
-
-
-%%%===================================================================
-%%% Internal functions
-%%%===================================================================
+extract_gets(#{op := get, args := [_, _,_, B, M]}) ->
+    {B, M}.
 
 
 %%--------------------------------------------------------------------
 %% @private
 %% @doc Translates the AST of the dql parser into a tree for dflow.
 %%
-%% @spec translate(DQLTerm :: term(),
-%%                 Aliases :: dict:dict(),
-%%                 Buckets :: orddict:orddict()) -> dflow:step().
 %% @end
 %%--------------------------------------------------------------------
-
-
-translate({calc, [], {get, {Bucket, Metric}}}, _Aliases, _Buckets) ->
-    {ok, {dqe_get, [Bucket, Metric]}};
+-spec translate(DQLTerm :: dql:query_part() | dql:dqe_fun()) ->
+                       {ok, pos_integer(), dflow:step()}.
```

Comment:
I was curious about the type spec here, because dql:dqe_fun() is defined as:
```
-type dqe_fun() ::
        #{
           op => fcall,
           signature => [sig_element()],
           args => #{}
         }.
```
I don't see a matching clause for such a definition, but the dialyzer does not give any such warnings.  What am I missing?